storage/fcds/meta.go

```diff
@@ -63,3 +67,18 @@ func (m *Meta) String() (s string) {
 	}
 	return fmt.Sprintf("{Size: %v, Offset %v}", m.Size, m.Offset)
 }
+
+type byVal []ShardInfo
+
+func (a byVal) Len() int           { return len(a) }
+func (a byVal) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a byVal) Less(i, j int) bool { return a[j].Val < a[i].Val }
+
+// ShardInfo contains data about an arbitrary shard
+// in that Val could potentially represent any scalar
+// size pertaining to a shard (number of free slots,
+// size in bytes, number of occupied slots, etc).
+type ShardInfo struct {
+	Shard uint8
+	Val   int64
+}
```

Comment:
Would not be better to be more precise on the Val? I find it quite strange that a value meaning is so vague. Could it be that we provide only the number of chunks as the value. That would be quite trivial to have in leveldb MetaStore implementation then the number of bytes.

Reply:
whoops. let me elaborate. originally due to the fact we were looking at the number of free slots in the shard _and_ the size of the shard, and we needed to sort both in the same way - i renamed this to be general in order to not keep two struct definitions (and two duplicate sorting primitives as of such). i see your point now that this is only used by `ShardSize`. i can rename it to conform :+1: